ion-schema/src/violation.rs

```diff
@@ -146,3 +153,59 @@ impl fmt::Display for ViolationCode {
         )
     }
 }
+
+#[cfg(test)]
+mod violation_tests {
+    use crate::ion_path::IonPath;
+    use crate::violation::{Violation, ViolationCode};
+
+    #[test]
+    fn violation_equivalence() {
+        let violation1: Violation = Violation::with_violations(
+            "type_constraint",
+            ViolationCode::TypeMismatched,
+            "type mismatched",
+            &mut IonPath::default(),
+            vec![
+                Violation::new(
+                    "regex",
+                    ViolationCode::RegexMismatched,
+                    "regex mismatched",
+                    &mut IonPath::default(),
+                ),
+                Violation::new(
+                    "container_length",
+                    ViolationCode::InvalidLength,
+                    "invalid length",
+                    &mut IonPath::default(),
+                ),
+            ],
+        );
+        let violation2: Violation = Violation::with_violations(
+            "type_constraint",
+            ViolationCode::TypeMismatched,
+            "type mismatched",
+            &mut IonPath::default(),
+            vec![
+                Violation::new(
+                    "container_length",
+                    ViolationCode::InvalidLength,
+                    "invalid length",
+                    &mut IonPath::default(),
+                ),
+                Violation::new(
+                    "regex",
+                    ViolationCode::RegexMismatched,
+                    "regex mismatched",
+                    &mut IonPath::default(),
+                ),
+            ],
+        );
+        assert_ne!(violation1.violations(), violation2.violations());
```

Comment:
question—why are we asserting that this are not equal? This seems like it's not relevant to the test.

Reply:
Err: this was part of debugging. I will remove this.